### PR TITLE
fix: astronomical loss with ModernBERT when using gradient checkpointing (#38982)

### LINKED ISSUE #38982: ModernBERT training loss is astronomical when using gradient accumulation
### System Info

- `transformers` version: 4.52.4
- Platform: Linux-6.6.87.1-microsoft-standard-WSL2-x86_64-with-glibc2.35
- Python version: 3.12.5
- Huggingface_hub version: 0.33.0
- Safetensors version: 0.5.3
- Accelerate version: 1.8.1
- Accelerate config:    not found
- DeepSpeed version: not installed
- PyTorch version (GPU?): 2.7.1+cu126 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using distributed or parallel set-up in script?: No
- Using GPU in script?: Yes
- GPU type: NVIDIA GeForce RTX 4090

### Who can help?

@SunMarc @ArthurZucker

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

1. Train ModernBERT on MLM with gradient accumulation.

### Expected behavior

Loss values should be the same as they would be without gradient accumulation. Instead, they get multiplied by gradient accumulation steps.

### PR DESCRIPTION
**Pull request by umarbutler**
@ArthurZucker @SunMarc This PR fixes #38982 by passing `kwargs` to `loss_function()` in the forward function for the ModernBERT MLM block -- no more astronomical loss when using gradient accumulation while training ModernBERT.

### CODE REVIEW

**Review by SunMarc — APPROVED** [review]
Indeed, thanks for fixing this ! I'm working on a PR that should help preventing this kind of issue in the future https://github.com/huggingface/transformers/pull/38432

**Comment by SunMarc** [comment]
@bot /style

**Comment by SunMarc** [comment]
Can you fix the CI ? You need to apply the modification to the files since you updated modular file with `make fix-copies`

**Review by ArthurZucker — APPROVED** [review]

**Comment by ArthurZucker** [comment]
@bot /style

**Review by ArthurZucker — COMMENTED** [review]
can you run the modular converter to update the modeling? 🤗

**Comment by umarbutler** [comment]
> can you run the modular converter to update the modeling? 🤗

How do I do that?

Given how minor this fix was, I had just edited it directly on GitHub.

**Comment by ArthurZucker** [comment]
No worries I'll take care of it!
